### PR TITLE
Add gateway installation

### PR DESCRIPTION
**Pull request by varungup90**
Address https://github.com/aibrix/aibrix/issues/114

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -160,10 +160,17 @@ endif
 .PHONY: install
 install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
 	$(KUSTOMIZE) build config/crd | $(KUBECTL) apply -f -
+    ## helm creates objects without aibrix prefix, hence deploying gateway components outside of kustomization	
+	helm install eg oci://docker.io/envoyproxy/gateway-helm --version v1.1.0 -n aibrix-system --create-namespace
```
Comment (Jeffwan) [inline Makefile]: line 161, `kustomize` should create the namespace. Did you see any problems missing the namespace here? We can leave the namespace here, I just like to confirm the sequence.

File: pkg/plugins/gateway/gateway.go [hunk 2]
```diff
@@ -183,7 +184,7 @@ func (s *Server) HandleRequestHeaders(ctx context.Context, req *extProcPb.Proces
 		}, user, targetPodIP
 	}
 
-	pods, err := s.client.CoreV1().Pods("default").List(ctx, v1.ListOptions{
+	pods, err := s.client.CoreV1().Pods(podutils.NAMESPACE).List(ctx, v1.ListOptions{
```
Comment (Jeffwan) [inline pkg/plugins/gateway/gateway.go:187]: Let's use injected env instead later. the default could be same

File: config/gateway/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+resources:
+- redis.yaml
+- gateway-plugin.yaml
```
Comment (Jeffwan) [inline config/gateway/kustomization.yaml:3]: should it include `gateway.yaml` here? or `gateway.yaml` has to be installed earlier?

File: Makefile [hunk 4]
```diff
@@ -160,10 +160,17 @@ endif
...
+	$(KUBECTL) wait --timeout=5m -n aibrix-system deployment/envoy-gateway --for=condition=Available
+	$(KUBECTL) apply -f config/gateway/gateway.yaml
```
Comment (Jeffwan) [inline Makefile:166]: em. Now we install kustomize package (gateway-plugin) -> dependencies -> gateway.yaml. The sequence is not that clean. Not sure any problem moving gateway.yaml under kustomize? We probably can do 3rd party dependencies (gateway) first, and then aibrix.

File: Makefile (hunk 1 above)
Comment (varungup90) [reply Makefile]: I will remove the `-create namespace`.

File: config/gateway/kustomization.yaml (hunk 3 above)
Comment (varungup90) [reply config/gateway/kustomization.yaml:3]: gateway.yaml is installed in `make install`. So here only gateway-plugin.yaml

File: Makefile (hunk 4 above)
Comment (varungup90) [reply Makefile:166]: make install does two things-> envoy gateway (using helm command) + gateway dependencies such as envoy patch policy & envoy extension policy (using gateway.yaml)

make deploy install redis and gateway plugins (ext proc).

File: pkg/plugins/gateway/gateway.go (hunk 2 above)
Comment (varungup90) [reply pkg/plugins/gateway/gateway.go:187]: for now will keep podutils.NAMESPACE and later move to injected env

Comment (Jeffwan) [reply pkg/plugins/gateway/gateway.go:187]: sounds good.

File: Makefile (hunk 4 above)
Comment (Jeffwan) [reply Makefile:166]: development flow might be different from one-click prod installation flow. If this is the flow. then we do not have `gateway.yaml` and `helm` package included. 
https://github.com/aibrix/aibrix/blob/0d64aca2c437b74fc64a4f0df9b6e81d7399ed95/README.md?plain=1#L16-L25

**Pull request merged**